Gems/ROS2/Code/Source/RobotImporter/URDF/VisualsMaker.cpp

```diff
@@ -100,99 +102,80 @@ namespace ROS2
         // Apply transform as per origin
         PrefabMakerUtils::SetEntityTransformLocal(visual->RawPose(), entityId);
 
-        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
         auto geometry = visual->Geom();
         switch (geometry->Type())
         {
         case sdf::GeometryType::SPHERE:
             {
                 auto sphereGeometry = geometry->SphereShape();
                 AZ_Assert(sphereGeometry, "geometry is not Sphere");
-                entity->CreateComponent(LmbrCentral::EditorSphereShapeComponentTypeId);
-                entity->Activate();
-                LmbrCentral::SphereShapeComponentRequestsBus::Event(
-                    entityId, &LmbrCentral::SphereShapeComponentRequests::SetRadius, sphereGeometry->Radius());
-                LmbrCentral::EditorShapeComponentRequestsBus::Event(
-                    entityId, &LmbrCentral::EditorShapeComponentRequests::SetVisibleInGame, true);
-                entity->Deactivate();
+                const AZ::Vector3 sphereDimensions(sphereGeometry->Radius());
+
+                // The `_sphere_1x1.azmodel` is created by Asset Processor based on O3DE `PrimitiveAssets` Gem source.
+                AZ::Data::AssetId assetId;
+                const char* sphereAssetRelPath = "objects/_primitives/_sphere_1x1.azmodel"; // relative path to cache folder.
```

Comment:
The Primitive Assets Gem should be added to the gem.json list of [dependencies](https://github.com/o3de/o3de-extras/blob/development/Gems/ROS2/gem.json#L27-L34).